plugins/grpc-transport-auth/src/main/java/com/github/bsideup/liiklus/transport/grpc/config/GRPCAuthConfiguration.java

```diff
@@ -39,40 +39,40 @@ public void initialize(GenericApplicationContext applicationContext) {
 
         log.info("GRPC Authorization ENABLED with algorithm {}", authProperties.getAlg());
 
-        // Init it early to check that everything is fine in config
-        JWTVerifier verifier = createVerifier(authProperties.getAlg(), authProperties);
-
         applicationContext.registerBean(
                 JWTAuthGRPCTransportConfigurer.class,
-                () -> new JWTAuthGRPCTransportConfigurer(verifier)
+                () -> new JWTAuthGRPCTransportConfigurer(authProperties)
         );
     }
 
-    private JWTVerifier createVerifier(GRPCAuthProperties.Alg alg, GRPCAuthProperties properties) {
-        switch (alg) {
-            case HMAC512:
-                return JWT
-                        .require(Algorithm.HMAC512(properties.getSecret()))
-                        .acceptLeeway(2)
-                        .build();
-            case RSA512:
-                return JWT
-                        .require(Algorithm.RSA512(new StaticRSAKeyProvider(properties.getKeys())))
-                        .acceptLeeway(2)
-                        .build();
-            default:
-                throw new IllegalStateException("Unsupported algorithm");
-        }
-    }
-
     @Value
     static class JWTAuthGRPCTransportConfigurer implements GRPCLiiklusTransportConfigurer {
-        private JWTVerifier verifier;
+
+        private GRPCAuthProperties properties;
 
         @Override
         public void apply(NettyServerBuilder builder) {
+            JWTVerifier verifier = createVerifier(properties.getAlg(), properties);
+
             builder.intercept(new JwtServerInterceptor<>(verifier::verify));
         }
+
+        static JWTVerifier createVerifier(GRPCAuthProperties.Alg alg, GRPCAuthProperties properties) {
```

Comment:
Probably doesn't need to be static anymore. Also, it is a bit weird that `alg` is passed but can be obtained from `GRPCAuthProperties` :)